packages/mui-system/src/Unstable_Grid/gridGenerator.ts

```diff
@@ -34,10 +34,20 @@ export const traverseBreakpoints = <T = unknown>(
   } else if (responsize && typeof responsize === 'object') {
     // prevent null
     // responsize could be a very big object, pick the smallest responsive values
-    const keys =
+    const keys = (
       Object.keys(responsize).length > breakpoints.keys.length
         ? breakpoints.keys
-        : Object.keys(responsize);
+        : Object.keys(responsize)
+    ).sort((key1, key2) => {
+      // @ts-ignore already checked that responsize is an object
+      const breakpointValue1: T = breakpoints.values[key1];
+      // @ts-ignore already checked that responsize is an object
+      const breakpointValue2: T = breakpoints.values[key2];
+      if (typeof breakpointValue1 === 'number' && typeof breakpointValue2 === 'number') {
+        return breakpointValue1 - breakpointValue2;
+      }
+      return 0;
+    });
```

Comment:
Can you create a util function in this file with a test to make sure that it works as expected?

Something like this:

```js
const keys = sortBreakpoints(breakpoints, responsive);
```

Reply:
Yeah sure I'll do it

Reply:
I like the fix from https://github.com/mui/material-ui/pull/34985/files, it looks simple. Can you do that with a test?

Reply:
ok sure, i'll do that and add test

Reply:
> I like the fix from https://github.com/mui/material-ui/pull/34985/files, it looks simple. Can you do that with a test?

i've added the test and rewrote logic to sort keys